packages/core/src/client/findSourceMap.ts

```diff
@@ -0,0 +1,86 @@
+// @ts-expect-error
+import { SourceMapConsumer } from 'source-map';
```

Comment:
I think import `source-map` will resolve to the Node.js version and get an error like this:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/aa567f79-b124-4d3a-9fea-ea1de27643b9)

Reply:
And we do not want to add third-party dependencies to `@rsbuild/core`, unless it can be prebundled to the `compiled` folder.

Reply:
Yes, I tried to upgrade `source-map` to 0.7.4 and ran into some problems. It seems to be a problem with the `source-map` package itself.

Do you think it would work if I prebundled `source-map@0.5.7` into the compiled directory? If yes, I'll try to do it this way

Reply:
However, version `0.7.4` has a major problem that makes it unusable on the browser side. I don't think `0.5.7` should be rejected just because it's outdated, but whether it's more important to consider its usability.

I have found no good solution to this problem without changing the version.  https://github.com/mozilla/source-map/issues/459

Reply:
We cannot guarantee the usability of a legacy version, it may contain some bugs and unsupported features.

Reply:
Is source-map necessary? What will be the impact if we don’t rely on source-map

Reply:
It is necessary to map source files through source-map.

If there is no such function, only error stack is displayed on overlay, which is greatly compromised in terms of experience. In my opinion, the important function of pr is lost.

Reply:
Maybe we can use [source-map-js](https://www.npmjs.com/package/source-map-js).

Reply:
> Maybe we can use [source-map-js](https://www.npmjs.com/package/source-map-js).

@9aoy Thanks for sharing. 

I have a new problem. When I prebundler, I get an error: `__filename is not defined`.

I didn't find a good solution and had to improvise a less elegant solution.

The esm polyfill for `__dirname` is mostly Node-based. Sincerely hope to find a better solution.

Reply:
The prebundle is used to bundle Node.js packages, and `source-map-js` in imported in the client code, so it can not be prebundled.

Reply:
For the client code, I think the new code added in the PR should be imported on demand. 

In other words, when the user does not enable the runtime error overlay, these code and `source-map-js` should be bundled into the client code, otherwise it will introduce unused code and slow down the build.

Reply:
> In other words, when the user does not enable the runtime error overlay, these code and `source-map-js` should not be bundled into the client code, otherwise it will introduce unused code and slow down the build.

Okay, let's take a look